server/src/main/java/org/opensearch/rest/action/cat/RestShardsAction.java

```diff
@@ -323,6 +324,7 @@ Table buildTable(RestRequest request, ClusterStateResponse state, IndicesStatsRe
             }
             table.addCell(shard.state());
             table.addCell(getOrNull(commonStats, CommonStats::getDocs, DocsStats::getCount));
+            table.addCell(getOrNull(commonStats, CommonStats::getDocs, DocsStats::getDeleted));
```

Comment:
Can we add it only if caller requests?

Reply:
The table entries are populated for all the fields (including non-default ones as well) irrespective of whatever columns are queried by the caller.